src/librustc/session/config.rs

```diff
@@ -2184,10 +2184,15 @@ pub fn build_session_options_and_crate_config(
         TargetTriple::from_triple(host_triple())
     };
     let opt_level = {
-        if matches.opt_present("O") {
-            if cg.opt_level.is_some() {
-                early_error(error_format, "-O and -C opt-level both provided");
+        let max_o = matches.opt_positions("O").into_iter().max();
```

Comment:
NB: `max` is not what the common expectation will be for these flags. Instead users expect that the last flag takes precedence.

Also for `opt-level` it is not possible to establish absolute ordering between `s`, `z` and the numeric levels.

Reply:
Ah never mind me, I read the code wrong and missed that this is taking the maximum of argument *position*.